### PR TITLE
feat: add force-api-refresh option to the api push command in order to support refreshing an API (for example, on resolver code changes)

### PR DESCRIPTION
**Pull request by alharris-at**
#### Description of changes
Adding support for the `--force` parameter to `gql-compile` which will ensure that the `schema.graphql` file has whitespace appended, updating the hash, and forcing a rebuild/push to backend. This is useful for adopting changes in generated resolvers/schema from the library, without requiring any codechange on the customer end.

##### CDK / CloudFormation Parameters Changed
N/A

#### Issue #, if available
N/A - internal ticket

#### Description of how you validated changes
Manually tested using `amplify-dev api gql-compile --force && amplify push`, and also validated that if make a destructive change (e.g. remove a model) it still blocks the push.

```
➜  testUpdate amplify-dev api gql-compile --force && amplify push
✅ GraphQL schema compiled successfully.

Edit your schema at /Users/alharris/apps/testUpdate/amplify/backend/api/testupdate/schema.graphql or place .graphql files in a directory at /Users/alharris/apps/testUpdate/amplify/backend/api/testupdate/schema
✔ Successfully pulled backend environment dev from the cloud.

    Current Environment: dev
    
┌──────────┬───────────────┬───────────┬───────────────────┐
│ Category │ Resource name │ Operation │ Provider plugin   │
├──────────┼───────────────┼───────────┼───────────────────┤
│ Api      │ testupdate    │ Update    │ awscloudformation │
└──────────┴───────────────┴───────────┴───────────────────┘
✔ Are you sure you want to continue? (Y/n) · yes
🛑 Removing a model from the GraphQL schema will also remove the underlying DynamoDB table.
This update will remove table(s) [Model2Table]
ALL EXISTING DATA IN THESE TABLES WILL BE LOST!
If this is intended, rerun the command with '--allow-destructive-graphql-schema-updates'.

Learn more at: https://docs.amplify.aws/cli/project/troubleshooting/

Session Identifier: ac2be8e7-9b15-4fa9-9eaf-28ca462c7d79
```

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-api/src/force-updates/api-resource-paths.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { pathManager, stateManager } from '@aws-amplify/amplify-cli-core';
```
Comment (alharris-at) [inline packages/amplify-category-api/src/force-updates/api-resource-paths.ts:1]: This file is just the existing functions from `auth-notifications.ts` which pertained to getting the api resource path.

File: packages/amplify-category-api/src/force-updates/force-refresh-schema.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import path from 'path';
+import fs from 'fs-extra';
+import { getApiResourceDir } from './api-resource-paths';
+
+export const forceRefreshSchema = (): void => {
```
Comment (alharris-at) [inline packages/amplify-category-api/src/force-updates/force-refresh-schema.ts:5]: This is just the functions from `auth-notifications.ts` which pertained to modifying the graphql schema, and exported as a new top-level method.

File: packages/amplify-category-api/src/graphql-transformer/transform-graphql-schema.ts [hunk 3]
```diff
@@ -4,6 +4,9 @@ import { $TSContext, ApiCategoryFacade } from '@aws-amplify/amplify-cli-core';
 import { AmplifyGraphQLTransformerErrorConverter } from '../errors/amplify-error-converter';
 import { transformGraphQLSchemaV1 } from './transform-graphql-schema-v1';
 import { transformGraphQLSchemaV2 } from './transform-graphql-schema-v2';
+import { forceRefreshSchema } from '../force-updates/force-refresh-schema';
+
+const FORCE_API_REFRESH_FLAG = 'force-api-refresh';
```
Comment (sobolk) [inline packages/amplify-category-api/src/graphql-transformer/transform-graphql-schema.ts:9]: Do we need new flag?
There's already `amplify push api --force` - should that just refresh schema?
See https://docs.amplify.aws/cli/commands/push/

**Comment by dpilch** [comment]
There is some error messaging in DataStore that will need to be updated when this is released. https://github.com/aws-amplify/amplify-js/blob/main/packages/datastore-storage-adapter/src/common/CommonSQLiteAdapter.ts#L92

File: packages/amplify-category-api/src/graphql-transformer/transform-graphql-schema.ts (hunk 3 above)
Comment (alharris-at) [reply packages/amplify-category-api/src/graphql-transformer/transform-graphql-schema.ts]: Updated to push `--force` to the `gql-compile` step, allowing us to keep from increasing the surface area on the options.

**Review by sundersc — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Pull request merged**